diff --git a/MediterraneanNight/gtk-2.0/apps/panel.rc b/MediterraneanNight/gtk-2.0/apps/panel.rc
index fd23793..8286a6a 100644
--- a/MediterraneanNight/gtk-2.0/apps/panel.rc
+++ b/MediterraneanNight/gtk-2.0/apps/panel.rc
@@ -1,12 +1,53 @@
+style "murrine-dark" {
+      # Stolen from https://bbs.archlinux.org/viewtopic.php?pid=1026506#p1026506
+      # I (rbrito) made the following substitutions:
+      # @text_color_dark with @theme_text_dark_color
+      # @bg_color_dark with @theme_bg_dark_color
+      base[NORMAL]		= @theme_bg_dark_color
+      base[PRELIGHT]		= shade (0.95, @theme_bg_dark_color)
+      base[INSENSITIVE]		= @theme_bg_dark_color
+      bg[NORMAL]		= @theme_bg_dark_color
+      bg[PRELIGHT]		= shade (1.02, @theme_bg_dark_color)
+      bg[SELECTED]		= shade (0.90, @selected_bg_color)
+      bg[INSENSITIVE]		= @theme_bg_dark_color
+      bg[ACTIVE]		= shade (1.04, @theme_bg_dark_color)
+      fg[NORMAL]		= @theme_text_dark_color
+      fg[PRELIGHT]		= @theme_text_dark_color
+      fg[SELECTED]		= @theme_text_dark_color
+      fg[ACTIVE]		= @theme_text_dark_color
+      fg[INSENSITIVE]		= darker (@bg_color)
+      text[NORMAL]		= @theme_text_dark_color
+      text[PRELIGHT]		= @theme_text_dark_color
+      text[ACTIVE]		= @theme_text_dark_color
+      text[SELECTED]		= @theme_text_dark_color
+      text[INSENSITIVE]		= darker (@theme_bg_dark_color)
+}
+
+# Stolen from: https://forum.xfce.org/viewtopic.php?pid=28950#p28950
+# I (rbrito) made the following substitutions:
+# @text_color_dark with @theme_text_dark_color
+# @bg_color_dark with @theme_bg_dark_color
+style "murrine-wide-dark"    = "murrine-dark"    {
+      xthickness = 2 ythickness = 2
+}
+
+style "murrine-menubar" = "murrine-dark" {
+    ythickness    = 0
+    bg[NORMAL]    = @bg_color
+    fg[NORMAL]    = @fg_color
+    fg[PRELIGHT]    = @fg_color
+    fg[SELECTED]    = @fg_color
+}
+
 style "theme-panel" = "murrine-dark"
 {
 	xthickness	= 0
 	ythickness	= 0
 
-	bg[NORMAL]	= shade (0.7, @bg_color_dark)
-	bg[ACTIVE]	= shade (0.5, @bg_color_dark)
-	bg[PRELIGHT]	= shade (0.35, @bg_color_dark)
-	bg[SELECTED]	= shade (0.5, @bg_color_dark)
+	bg[NORMAL]	= shade (0.7, @theme_bg_dark_color)
+	bg[ACTIVE]	= shade (0.5, @theme_bg_dark_color)
+	bg[PRELIGHT]	= shade (0.35, @theme_bg_dark_color)
+	bg[SELECTED]	= shade (0.5, @theme_bg_dark_color)
 
 	fg[NORMAL]	= shade (0.9, @base_color)
 	fg[PRELIGHT]	= @base_color
@@ -42,6 +83,10 @@ style "theme-panel-text"
 
 style "panel-entry" = "murrine-dark"
 {
+        bg[SELECTED]	= shade(1.1, @selected_bg_color)
+        bg[ACTIVE]	= shade(1.1, @selected_bg_color)
+        bg[NORMAL]	= shade(1.1, @selected_bg_color)
+
 	fg[NORMAL]	= @text_color
 	fg[PRELIGHT]	= @text_color
 	fg[ACTIVE]	= @text_color
@@ -93,9 +138,9 @@ class "*Xfce*Panel*"                style "theme-panel"
 widget "*WnckPager*"	    	    style "workspace-switcher"
 
 # Fix gtk-entries in the panel
-class "GtkEntry"		    style "murrine-combo"
-class "*SexyIconEntry*"		    style:highest "murrine-combo"	# fixes dict-plugin
-widget "*xfce4-verve-plugin*GtkEntry"	style:highest "murrine-combo"	# fixes verve-plugin
+class "GtkEntry"		    style "entry"
+class "*SexyIconEntry*"		    style:highest "entry"	# fixes dict-plugin
+widget "*xfce4-verve-plugin*GtkEntry"	style:highest "entry"	# fixes verve-plugin
 
 # Make sure panel text color doesn't change
 widget_class "*Panel*MenuBar*"      style "theme-main-menu-text"
@@ -104,12 +149,12 @@ widget "*.clock-applet-button.*"    style "theme-panel-text"
 widget "*PanelApplet*"              style "theme-panel-text"
 
 # dark menus
-widget_class "*<GtkMenu>*"              	style "murrine-dark"
+widget_class "*<GtkMenu>*"              	style "theme-main-menu-text"
 widget_class "*<GtkMenuItem>*"          	style "murrine-wide-dark"
 widget_class "*<GtkMenuItem>.*.<GtkCalendar>"	style "murrine-dark"
 widget_class "*<GtkMenuItem>.*.<GtkEntry>"	style "panel-entry"
-widget_class "*<GtkMenuItem>.*.<GtkScale>" 	style "murrine-scale-dark"
-widget_class "*<GtkMenuBar>*"           	style "murrine-menubar"
+widget_class "*<GtkMenuItem>.*.<GtkScale>" 	style "murrine-dark"
+widget_class "*<GtkMenuBar>*"           	style "theme-main-menu-text"
 widget "*.gtk-combobox-popup-menu.*"   		style "murrine-wide-dark"
 
 # dark bg and black text for gdm's panel
diff --git a/MediterraneanNight/gtk-2.0/apps/terminal.rc b/MediterraneanNight/gtk-2.0/apps/terminal.rc
index aa48c5f..bbf34be 100644
--- a/MediterraneanNight/gtk-2.0/apps/terminal.rc
+++ b/MediterraneanNight/gtk-2.0/apps/terminal.rc
@@ -13,28 +13,14 @@ style "terminal-scrollbar" = "scrollbar"
 
 }
 
+# SELECTED means the "current" widget; ACTIVE means at the moment of click
+
 style "terminal-notebook"
 {
-	bg[NORMAL]		= shade( 0.85, @sidebar_background)
-	bg[ACTIVE]		= shade( 0.70, @sidebar_background)
+	bg[NORMAL]	= shade(1.1, @bg_color)
+	bg[ACTIVE]	= shade(0.875, @bg_color)
 	fg[NORMAL] 	= shade(0.80, @text_color) # text selected tab
 	fg[ACTIVE] 	= shade(0.28, @sidebar_background)	# text unselected tab
-
-	engine "murrine"
-	{
-		gradient_shades		= {1.00,0.96,0.92,0.88}
-		highlight_shade     = 1.03
-		roundness       	= 0
-		contrast        	= 0.0	
-		
-		## things that don't work:
-		#glow_shade	= 0.8
-		#glowstyle           = 2
-		#focusstyle	    = 3
-		#glazestyle          = 3
-		#border_shades	= { 0.45, 2.5 }
-		#prelight_shade	= 1.5
-	}
 }
 
 style "notebook-button" {
@@ -42,8 +28,7 @@ style "notebook-button" {
     ythickness		= 0
     bg[ACTIVE]		= shade (0.68, @sidebar_background)
     bg[PRELIGHT]        = shade (0.75, @sidebar_background)
-	engine "murrine"
-	{
+    engine "murrine" {
 	gradient_shades		= {1.00,1.00,1.00,1.00}
 	contrast        	= 0.6
 	roundness       	= 0
diff --git a/MediterraneanNight/gtk-2.0/gtkrc b/MediterraneanNight/gtk-2.0/gtkrc
index d72350e..0d17b00 100644
--- a/MediterraneanNight/gtk-2.0/gtkrc
+++ b/MediterraneanNight/gtk-2.0/gtkrc
@@ -1046,4 +1046,4 @@ widget_class "*XfdesktopIconView*" style "xfdesktop-icon-view"
 include "apps/thunar.rc"		# thunar especific
 include "apps/terminal.rc"
 include "apps/gmusicbrowser.rc"
-
+include "apps/panel.rc"
